qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -624,26 +646,45 @@ class Gaussian(metaclass=_PulseType):
     def __new__(
         cls,
         duration: Union[int, ParameterExpression],
-        amp: Union[complex, ParameterExpression],
+        amp: Union[complex, float, ParameterExpression],
         sigma: Union[float, ParameterExpression],
+        angle: Optional[Union[float, ParameterExpression]] = None,
         name: Optional[str] = None,
         limit_amplitude: Optional[bool] = None,
     ) -> SymbolicPulse:
         """Create new pulse instance.
 
         Args:
             duration: Pulse length in terms of the sampling period `dt`.
-            amp: The amplitude of the Gaussian envelope.
+            amp: The magnitude of the amplitude of the Gaussian envelope. Complex amp support will be deprecated.
             sigma: A measure of how wide or narrow the Gaussian peak is; described mathematically
                    in the class docstring.
+            angle: The angle of the complex amplitude of the Gaussian envelope. Default value 0.
             name: Display name for this pulse envelope.
             limit_amplitude: If ``True``, then limit the amplitude of the
                 waveform to 1. The default is ``True`` and the amplitude is constrained to 1.
 
         Returns:
             SymbolicPulse instance.
         """
-        parameters = {"amp": amp, "sigma": sigma}
+        # This should be removed once complex amp support is deprecated.
+        if isinstance(amp, complex):
+            if angle is None:
+                warnings.warn(
+                    "Complex amp will be deprecated. Use float amp (for the magnitude) and float angle instead.",
+                    PendingDeprecationWarning
+                )
+            else:
+                raise PulseError("amp can't be complex when providing angle")
+
+        if isinstance(amp, float):
+            if amp < 0:
+                raise PulseError("'amp' has to be positive (use 'angle' to set sign)")
```

Comment:
To me this seems too strict and indeed this causes breaking API change, i.e. in calibration we often set negative amplitude for negative rotation. I would take abs and set pi to angle with `UserWarning`. In the same way, you can write something like
```python
def phasor_formatter(amp, angle)
    magnitude = np.abs(amp)
    angle = angle + np.angle(amp)
    return magnitude, angle
```

Reply:
I wasn't sure about this as well. My issue was the `amp_angle` property. The way I defined it, it returns
`return np.abs(self._params["amp"]),np.angle(self._params["amp"])`
This means that when you retrieve (amp,angle) - amp will always be positive. In turn, this means that if the user creates a pulse with
`Gaussian(duration=25, sigma=4, amp=-0.5, angle = 0)`
the (amp,angle) tuple he'll get is (0.5,pi). This is what led me to restrict to positive amp - to avoid this scenario where you plug in some (amp,angle) and retrieves a different set.

In my eyes, your proposal leaves the same problem. Maybe I am over-reacting here and this isn't really an issue.

Another way to solve both issues is to store the complex amp for backward compatibility like we do now, while also storing the (amp,angle) duo. In this scenario, if the user provides (amp,angle) we can just plug them in as is, while if he provides complex amp, we break it into (amp,angle).

I could be overthinking this. Let me know what you think.

Reply:
When you write a calibration/characterization experiment, you may define, for example, `x90p` pulse with the parameterized amplitude and then scan its amplitude [from negative to positive value](https://github.com/Qiskit/qiskit-experiments/blob/4cded9846971533372e8b45d50c9ed60bfe8ea46/qiskit_experiments/library/characterization/rabi.py#L85) to find the amplitude that gives you pi/2 rotation. You may also define another pulse, `y90p`, with shared amplitude, sigma, and duration with `angle=pi/2`. So whatever the sign of `x90p.amp` is, the `y90p` instance keeps pi/2 angle difference from the `x90p`. As this example explains, the domain of amp and angle should be real. This is the request from the application side, while in the transfer layer we should conform to the complex representation. Because the conversion doesn't depend on the sign of amplitude, i.e. `amp, angle = 0.1, pi` and `amp, angle = -0.1, 0.0` yield the same pulse, we should not limit the amplitude to be positive.

After the deprecation period, we must remove property method to return a particular pulse parameter. This means, when you instantiate the pulse with `(amp, angle) = (-0.1, pi/2)` you will get `my_pulse.amp = -0.1` and `my_pulse.angle = pi/2`. This makes me think storing amp and angle as-is in the parameters is the one of reasonable approaches. Probably, you can do
```python
def Gaussian(..., angle: Optional[float] = None):
    if angle is None and isinstance(amp, complex):
        warn("...", PendingDeprecationWarning)
    elif angle is None:
        angle = 0.0
```